### PR TITLE
django_manage: deprecate venv creation when missing

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
As pointed out in https://github.com/ansible-collections/community.general/issues/5374#issuecomment-1287536369
creating the virtualenv from within `django_manage` is not only a bad design decision but it simply does not work as it is.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
django_manage

### CODE REVIEW

File: changelogs/fragments/5404-django-manage-venv-deprecation.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+deprecated_features:
+  - >-
+    django_manage - The behavior of "creating the virtual environment when missing"
```
Comment (felixfontein) [inline changelogs/fragments/5404-django-manage-venv-deprecation.yml:3]:
```suggestion
    django_manage - the behavior of "creating the virtual environment when missing"
```

File: plugins/modules/web_infrastructure/django_manage.py [hunk 2]
```diff
@@ -109,11 +109,21 @@
     type: str
     required: false
     aliases: [test_runner]
+  ack_venv_creation_deprecation:
+    description:
+      - >-
+        When a I(virtualenv) is set but the virtual environment does not exist, the current behavior is
+        to create a new virtual environment. That behavior is deprecated and if that case happens it will
+        generate a deprecation warning. Set this flag to C(true) to suppress the deprecation warning.
+    type: bool
```
Comment (felixfontein) [inline plugins/modules/web_infrastructure/django_manage.py:118]:
```suggestion
    type: bool
    version_added: 5.8.0
```

File: plugins/modules/web_infrastructure/django_manage.py [hunk 3]
```diff
@@ -109,11 +109,21 @@
     type: str
     required: false
     aliases: [test_runner]
+  ack_venv_creation_deprecation:
+    description:
+      - >-
+        When a I(virtualenv) is set but the virtual environment does not exist, the current behavior is
+        to create a new virtual environment. That behavior is deprecated and if that case happens it will
+        generate a deprecation warning. Set this flag to C(true) to suppress the deprecation warning.
```
Comment (felixfontein) [inline plugins/modules/web_infrastructure/django_manage.py:117]:
```suggestion
        generate a deprecation warning. Set this flag to C(true) to suppress the deprecation warning.
      - Please note that you will receive no further warning about this being removed until the module
        will start failing in such cases from community.general 9.0.0 on.
```

File: plugins/modules/web_infrastructure/django_manage.py [hunk 4]
```diff
@@ -179,10 +189,23 @@ def _ensure_virtualenv(module):
...
         return
 
+    ack_venv_deprecation = module.params['ack_venv_creation_deprecation']
+
     vbin = os.path.join(venv_param, 'bin')
     activate = os.path.join(vbin, 'activate')
 
     if not os.path.exists(activate):
+        # In version 9.0.0, if the venv is not found, it should fail_json() here.
+        if not ack_venv_deprecation:
```
Comment (felixfontein) [inline plugins/modules/web_infrastructure/django_manage.py:199]: Why not use
```suggestion
        if not module.params['ack_venv_creation_deprecation']:
```
here and remove the above assignment? That makes it easier to remove this in 9.0.0.

Comment (russoz) [reply plugins/modules/web_infrastructure/django_manage.py]: Done

**Comment by felixfontein** [comment]
The remaining CI failures are unrelated.

**Comment by felixfontein** [comment]
@russoz thanks for your contribution!

**Pull request merged**